pkg/registry/core/adapters/ns_registry.go

```diff
@@ -34,16 +35,28 @@ type networkServiceRegistryServer struct {
 }
 
 func (n *networkServiceRegistryServer) Register(ctx context.Context, request *registry.NetworkService) (*registry.NetworkService, error) {
-	return n.client.Register(ctx, request)
+	doneCtx := withDone(ctx)
+	service, err := n.client.Register(doneCtx, request)
+	if err != nil {
+		return nil, err
+	}
+	if request == nil {
+		request = &registry.NetworkService{}
+	}
+	if !isDone(doneCtx) {
+		return service, nil
+	}
+	return next.NetworkServiceRegistryServer(ctx).Register(ctx, request)
 }
 
 func (n *networkServiceRegistryServer) Find(query *registry.NetworkServiceQuery, s registry.NetworkServiceRegistry_FindServer) error {
-	client, err := n.client.Find(s.Context(), query)
+	doneCtx := withDone(s.Context())
+	client, err := n.client.Find(doneCtx, query)
 	if err != nil {
 		return err
 	}
 
-	for {
+	for !isDone(doneCtx) {
```

Comment:
```suggestion
        if !isDone(doneCtx) {
            return nil
        }
	for {
```

Reply:
Also need to apply this fix  for nse_registry

Reply:
problem fixed another way